src/controllers/auth.controller.js

```diff
@@ -23,34 +23,27 @@ const RefreshToken = db.RefreshToken;
  */
 function login(req, res, next) {
     const params = _.pick(req.body, 'username', 'password');
-    const user = User.findOne({ where: { username: params.username } });
-    const passwordMatch = user.then((userResult) => {
-        if (_.isNull(userResult)) {
-            const err = new APIError('Incorrect username or password', 'INCORRECT_USERNAME_OR_PASSWORD', httpStatus.NOT_FOUND, true);
-            return next(err);
-        }
-        return userResult.testPassword(params.password);
-    });
 
-    // once the user and password promises resolve, send the token or an error
-    Promise.join(user, passwordMatch, (userResult, passwordMatchResult) => {
-        if (!passwordMatchResult) {
+    User.findOne({ where: { username: params.username } }).then((user) => {
```

Comment:
Should (I really don't know) this be `return User.findOne` like `function submitRefreshToken`: https://github.com/amida-tech/amida-auth-microservice/blob/4f1712b9810e5c795993ccf73125948d0d657ca6/src/controllers/auth.controller.js#L78

Might cause an edge-cause bug, where `next()` doesn't trigger and the default return from `function login` is `undefined`.

Reply:
Good question.

The fact that `submitRefreshToken()` returns an unresolved promise is actually useless. Express doesn't do anything with that unresolved promise.

What tells express to send an HTTP response is that `res.send()` is called somewhere by the controller (or one of its callbacks; or one of its promise resolution/rejection handlers).

I was going to type up a more thorough explanation, but that would take too much time. If you have further questions, feel free to LMK.